wear/src/main/java/io/homeassistant/companion/android/complications/EntityStateDataSourceService.kt

```diff
@@ -79,7 +79,14 @@ class EntityStateDataSourceService : SuspendingComplicationDataSourceService() {
         } else {
             null
         }
-        val text = PlainComplicationText.Builder(entity.friendlyState(this, entityOptions)).build()
+
+        var stateText = entity.friendlyState(this, entityOptions)
+        if (entity.attributes.containsKey("unit_of_measurement")) {
+            stateText += " " + entity.attributes["unit_of_measurement"]
+        }
+
+        val text = PlainComplicationText.Builder(stateText).build()
```

Comment:
For 2 maybe something like :point_down: Happy to get help with writing idiomatic Kotlin, I'm writing Kotlin for the first time in my life (using Java professionally for the past 10+ years though) . :grin:

```diff
diff --git common/src/main/java/io/homeassistant/companion/android/common/data/integration/Entity.kt common/src/main/java/io/homeassistant/companion/android/common/data/integration/Entity.kt
index 2a1bbc00..5ec0fd7e 100644
--- common/src/main/java/io/homeassistant/companion/android/common/data/integration/Entity.kt
+++ common/src/main/java/io/homeassistant/companion/android/common/data/integration/Entity.kt
@@ -652,6 +652,21 @@ fun <T> Entity<T>.friendlyState(context: Context, options: EntityRegistryOptions
     return friendlyState
 }
 
+fun <T> Entity<T>.friendlyStateWithOptionalUnit(
+    context: Context,
+    options: EntityRegistryOptions? = null
+): String {
+    val friendlyState = friendlyState(context, options)
+
+    val unit = (attributes as? Map<*, *>)?.get("unit_of_measurement")?.toString()
+
+    return if (unit?.isNotEmpty() == true) {
+        "$friendlyState $unit"
+    } else {
+        friendlyState
+    }
+}
+
 fun <T> Entity<T>.canSupportPrecision() = domain == "sensor" && state.toDoubleOrNull() != null
 
 fun <T> Entity<T>.isExecuting() = when (state) {
diff --git wear/src/main/java/io/homeassistant/companion/android/complications/EntityStateDataSourceService.kt wear/src/main/java/io/homeassistant/companion/android/complications/EntityStateDataSourceService.kt
index c2757f76..58287714 100644
--- wear/src/main/java/io/homeassistant/companion/android/complications/EntityStateDataSourceService.kt
+++ wear/src/main/java/io/homeassistant/companion/android/complications/EntityStateDataSourceService.kt
@@ -19,7 +19,7 @@ import dagger.hilt.android.AndroidEntryPoint
 import io.homeassistant.companion.android.common.R
 import io.homeassistant.companion.android.common.data.integration.canSupportPrecision
 import io.homeassistant.companion.android.common.data.integration.friendlyName
-import io.homeassistant.companion.android.common.data.integration.friendlyState
+import io.homeassistant.companion.android.common.data.integration.friendlyStateWithOptionalUnit
 import io.homeassistant.companion.android.common.data.integration.getIcon
 import io.homeassistant.companion.android.common.data.servers.ServerManager
 import io.homeassistant.companion.android.database.wear.EntityStateComplicationsDao
@@ -80,12 +80,12 @@ class EntityStateDataSourceService : SuspendingComplicationDataSourceService() {
             null
         }
 
-        var stateText = entity.friendlyState(this, entityOptions)
-        if (entity.attributes.containsKey("unit_of_measurement")) {
-            stateText += " " + entity.attributes["unit_of_measurement"]
-        }
-
-        val text = PlainComplicationText.Builder(stateText).build()
+        val text = PlainComplicationText.Builder(
+            entity.friendlyStateWithOptionalUnit(
+                this,
+                entityOptions
+            )
+        ).build()
 
         val contentDescription = PlainComplicationText.Builder(getText(R.string.complication_entity_state_content_description)).build()
         val monochromaticImage = MonochromaticImage.Builder(Icon.createWithBitmap(iconBitmap)).build()
```

Reply:
1. I agree, as some units can be quite long.
2. It would probably be nice to add it to a function instead of only here, so it can be used elsewhere. Combined with 1 an argument for the existing function (defaulting to false) is probably the easiest, allowing you to pass in the stored value instead of switching between two functions.

Reply:
Both 1 and 2 have been fixed now, please take a new look whenever you have time.

Reply:
Looks good!